backend/src/routes/api/segment-key/segmentKeyUtils.ts

```diff
@@ -3,14 +3,20 @@ import { KubeFastifyInstance, ODHSegmentKey } from '../../../types';
 export const getSegmentKey = async (fastify: KubeFastifyInstance): Promise<ODHSegmentKey> => {
   const coreV1Api = fastify.kube.coreV1Api;
   const namespace = fastify.kube.namespace;
+  let segmentKeyEnabled = true;
   try {
+    const resEnabled = await coreV1Api.readNamespacedConfigMap(
+      'rhods-segment-key-enabled',
+      namespace,
+    );
+    segmentKeyEnabled = resEnabled?.body.data?.segmentKeyEnabled === 'true';
     const res = await coreV1Api.readNamespacedSecret('rhods-segment-key', namespace);
     const decodedSegmentKey = Buffer.from(res.body.data.segmentKey, 'base64').toString();
```

Comment:
Is there any point in getting the segment key if it is not enabled?

Reply:
Yeah, you are completly right, updated!